instruqt/proxy/02-kurl-install/setup-isolated-host

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+set -euxo pipefail
+
+echo "Running challenge 02 setup script - Started.."
+epoc_start=`date +%s`
+
+# Determine OS platform
+CENTOS=false
+UBUNTU=false
+if [ -f /usr/bin/python ] ; then
+  if python -mplatform |grep -i 'centos'; then CENTOS=true; fi
+fi
+if uname -a | grep -i Ubuntu ; then UBUNTU=true; fi
+
+if $CENTOS ; then
+  NET_INTERFACE=eth0
+fi
+if $UBUNTU ; then
+  NET_INTERFACE=ens4
+fi
+
+#
+LOCAL_PROXY_IP=`host proxy-host | grep has | awk '{print $4}'`
+LOCAL_IP=`ifconfig $NET_INTERFACE | grep inet |grep -v inet6 | head -1 | awk '{print $2}'`
+LOCAL_SUBNET=`echo $LOCAL_IP | awk -F"." '{printf("%s.%s.0.0/16"),$1,$2}'`
+
+NO_PROXY="localhost,lo,.internal,.local,.svc,.default,169.254.169.254,127.0.0.0/24,kubernetes,10.0.0.0/8,10.5.0.0/16,10.2.0.0/16,10.32.0.0/20,10.96.0.0/12,10.96.0.10,$LOCAL_IP,$LOCAL_SUBNET,$LOCAL_PROXY_IP"
+
+#
+# systemwide proxy
+cat <<EOF> /etc/environment
+http_proxy=${LOCAL_PROXY_IP}:3128
+https_proxy=${LOCAL_PROXY_IP}:3128
+no_proxy=$NO_PROXY
+EOF
+#
+# local shell environment
+echo "export LOCAL_PROXY_IP=$LOCAL_PROXY_IP" >> /root/.bashrc
+echo "export http_proxy=http://${LOCAL_PROXY_IP}:3128" >> /root/.bashrc
+echo "export https_proxy=http://${LOCAL_PROXY_IP}:3128" >> /root/.bashrc
+echo "export no_proxy=$NO_PROXY" >> /root/.bashrc
```

Comment:
Isn't this something the end-user should do as part of the instructions?

Reply:
I thought about that, but encountered some issues with the kots app on kurl when the proxy url used the instruqt host short name.  IP address was reliable so wanted to explicitly use that in the variables to minimise lab failures.  Also NO_PROXY is very sensitive to it's contents, way too much to go into in the lab, I should add a link to the docs on this though as will be different for a real end customer env with a working DNS!